lib/lambda.js

```diff
@@ -559,10 +594,57 @@ function wrapNestedPath (agent, field, module) {
   }
 }
 
+function lowerCaseObjectKeys (obj) {
+  const lowerCased = {}
+  for (const key of Object.keys(obj)) {
+    lowerCased[key.toLowerCase()] = obj[key]
+  }
+  return lowerCased
+}
+
+// Reduce a SQS record message attributes object to a headers-like object of
+// string key/value-pairs. Some attribute types are ignored.
+// https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html#message-attribute-data-types
+//
+// `record.messageAttributes` is of the form:
+//    { <attribute-name>: { dataType: <attr-type>, stringValue: <attr-value>, ... } }
+// For example:
+//    { traceparent: { dataType: 'String', stringValue: 'test-traceparent' } }
+function headersFromSqsMessageAttributes (messageAttributes) {
```

Comment:
👍 I see a new `headersFromSqsMessageAttributes` that simplifies the data from amazon's strong-ishly typed JSON data into something that APM Server can handle. (I also see a separate function for SNS messages)